packages/ui/src/primitives/spinner.tsx

```diff
@@ -0,0 +1,135 @@
+import cn from 'clsx';
+import { forwardRef, type HTMLAttributes } from 'react';
+
+const SIZE = 16;
+
+type SpinnerRef = HTMLDivElement;
+
+/**
+ * # Spinner
+ *
+ * ## Accessibility
+ *
+ * An accessible label–set via `children`–must be provided for screen readers.
+ *
+ * ```tsx
+ * import { Spinner } from '@clerk/ceramic/components/Spinner';
+ *
+ * function Example() {
+ *   return (
+ *     <Spinner>
+ *       Some other loading text…
+ *     </Spinner>
+ *   );
+ * }
+ * ```
+ */
+export const Spinner = forwardRef(function Spinner(
+  {
+    children,
+    className,
+  }: Pick<HTMLAttributes<SpinnerRef>, 'className'> & {
+    children: string;
+  },
+  ref: React.ForwardedRef<SpinnerRef>,
+) {
+  return (
+    <span
+      ref={ref}
+      className={cn('relative isolate text-current size-[1em] block', className)}
```

Comment:
@joe-bell added block here as the size value isnt respected for inline elements.